### DISCUSSION TITLE
[November 2024] ZKsync monthly developer update

### DISCUSSION THREAD

**Discussion by uF4No** [zkSync-Community-Hub/zksync-developers#821] [Announcements]
Welcome to the November Developer Update! This month has been packed with exciting milestones, events, and continuous improvements to our ecosystem. Here’s a breakdown of what’s new:

### **ZKSync SSO Launch**  

![zksync-sso-dark](https://github.com/user-attachments/assets/8b14138f-2083-40ef-9aba-0d800f55f188)

We are thrilled to announce the launch of **ZKSync SSO (Smart Sign-On)**, a game-changer for integrating modular smart accounts into web applications. This SDK is designed to revolutionize user experiences by offering:  

- **Modules**: Smart accounts based on ERC7579.
- **Passkeys**: Biometric authentication for seamless access.  
- **Sessions**: Say goodbye to signing each transaction manually.  
- **Spending Limits**: Enhanced security through controlled transaction parameters.  

Dive deeper into the [documentation](https://docs.zksync.io/zksync-era/unique-features/zksync-sso) or [try the demo](https://nft.zksync.dev/) to experience it firsthand.

### **DEVCON[7] Bangkok**  
<img width="960" alt="devcon" src="https://github.com/user-attachments/assets/2e37575b-219c-449d-a8f9-64f15f09ba64" />

Our team had an incredible time at DEVCON in Bangkok, participating in multiple panels, talks and workshops. Here’s a quick recap of our contributions:  

- *The Rise of Appchains: From L2s to Rollup Clusters*  : [Watch here](https://www.youtube.com/watch?v=HHl2iOgP4FA)  
- *I Read Every Single 1990s Cypherpunk Email. Here’s What You Should Know*  [Watch here](https://youtu.be/4DtB96PlAtQ?feature=shared)  
- *Improving the User Experience by User Research*  [Watch here](https://www.youtube.com/watch?v=1PpaGPkCfpY)  
- *Bringing South America On-Chain with SSI and Account Abstraction*  [Watch here](https://www.youtube.com/watch?v=5c43HjCcZeg)  

Check out all the talks on the [Ethereum Foundation YouTube channel](https://www.youtube.com/@EthereumFoundation/videos).

### **Developer Tools Updates**  

We continue enhancing developer tools to improve functionality and ease of use:  

- **Hardhat**:  
  - Fixed contract verification for contracts with constructor parameters.  
  - Added support for using paymasters during contract upgrades.  

- **zksync-ethers**:  
  - Updated dependencies to patch vulnerabilities.  
  - Switched to `create2Factory` for deploying contracts, ensuring deterministic addresses across accounts.  

### **Block Explorer Updates**  

Our block explorer received several improvements this month to enhance usability:  

- Verified contracts now feature a **`Source Code` badge** and a checkmark in the Contracts tab.  
- ABIs of verified contracts can now be viewed and copied effortlessly.  
- Batch details are grouped for better readability.  
- Added **`View All Blocks`** and **`View All Transactions`** buttons on the landing page for quicker navigation.  

These update are live, check them out on [the ZKsync Era explorer](https://explorer.zksync.io/).

### **Cyfrin Python and Vyper Courses**  

![CleanShot 2024-12-10 at 16 42 14](https://github.com/user-attachments/assets/11f43410-d730-47a1-8265-42a14c47aa68)

The Cyfrin Updraft team released new courses on **Python** and **Vyper**, as always free for everyone and showcasing how to build on ZKsync Era. Check them out [here](https://x.com/CyfrinUpdraft/status/1861778155284922491).  

### **Ignite: Fueling Liquidity on ZKSync Era**  

The Ignite program aims to transform ZKSync Era into the liquidity hub of the Elastic Chain. Following a successful on-chain vote, **300M $ZK** has been allocated to incentivize DeFi protocols starting January. Learn more on the official [Ignite website](https://zksyncignite.xyz/).

### **New Podcast Episode: Perks of the Perps**  

Our latest podcast episode, *Perks of the Perps*, is live! Catch it on [YouTube](https://www.youtube.com/watch?v=Bkn6COmK6tQ) for insights into perpetual swaps and their significance in the ecosystem.

###  Stay connected

- Follow [[@zkSyncDevs](https://twitter.com/zkSyncDevs)](https://twitter.com/zkSyncDevs) on X
- Join the #dev-general channel in [[our Discord](https://join.zksync.dev/)](https://join.zksync.dev/)

That wraps up November’s highlights. As always, thank you for being a part of the ZKSync journey. We can’t wait to see what we accomplish together next month! 🎉

Keep buidling! 👩‍💻🧑🏻‍💻

**Comment by bxpana** [comment]
These are great! Thanks for taking the time to put them together and keep things succinct 😁
